pkgs/by-name/pr/protege/package.nix

```diff
@@ -0,0 +1,93 @@
+{ lib
+, fetchFromGitHub
+, copyDesktopItems
+, iconConvTools
+, makeDesktopItem
+, makeWrapper
+, jdk11
+, maven
+}:
+
+let
+  mvn = maven.override { jdk = jdk11; };
+
+  pname = "protege";
+  version = "5.6.4";
```

Comment:
Also, personally, I don't think that using `let in` just to avoid a `rec` is so required here, as all of these attributes are used eventually in the final function arguments. `let in` is useful if you have many computations to perform with Nix, and you don't want tiny Nix variable renamings to affect the environment of the build. Here, it is not the case.

I'd even go as far as:

```nix
(maven.override { jdk = jdk11; }).buildMavenPackage rec {
  pname = "proegege";
  version = "5.6.4";
  # ...

  meta = {# ...
  };
}

Reply:
I agree, but I think I'll leave the Maven override inside of the let-bind. I feel like an in-line override at the head of a line would add too much noise and distract many reading eyes (mine included).